cylc/flow/task_state.py

```diff
@@ -349,8 +349,11 @@ def suicide_prerequisites_all_satisfied(self):
 
     def prerequisites_get_target_points(self):
         """Return a list of cycle points targeted by each prerequisite."""
-        return set(point for prerequisite in self.prerequisites for
-                   point in prerequisite.get_target_points())
+        return {
+            point
+            for prerequisite in self.prerequisites
+            for point in prerequisite.get_target_points()
+        }
```

Comment:
That would be an error:

```python
>>> prereqs = [[1]]
>>> {point for prereq in prereqs for point in prereq}
{1}
>>> {point for point in prereq for prereq in prereqs}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'prereq' is not defined
```

Some explanation on this SO https://stackoverflow.com/a/38321692

Note the Python way the comprehension order is the same as it would be for a nested for loop:

```python
# list comp
return {
    point
    for point in prerequisite.get_target_points()
    for prerequisite in self.prerequisites
}

# for loop
for point in prerequisite.get_target_points():
    for prerequisite in self.prerequisites:
        points.append(point)
```

Reply:
Ugh, clearly I need to unlearn a bad habit I've picked up from somewhere